posit-dev/positron
# [python] log `positron_ipykernel` warnings, do not show to user

### Intent

> Describe briefly what problem this pull request resolves, or what new feature it introduces. Include screenshots of any new or altered UI. Link to any GitHub issues that are related.

related to #2615 #2446 (sort of #2075)

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

Using a custom handler for warnings. It checks if the warning is coming from `python_files/positron/positron_ipykernel` and then routes it to the logs. If the error is coming from the console, do not show temp filename. You can still override this filter, eg, if you would like to raise DeprecationWarnings in the console, etc.

Video of what warnings should look like ⏬ 

https://github.com/posit-dev/positron/assets/54685329/c05748aa-d0be-42bc-8202-9204eed35f21

todo

- add tests
- windows paths?

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues.

```python
import warnings

warnings.warn("this is a warning")
```
should not see temp path before warning (something like var/folders/.../47329.py)
